### ISSUE TITLE
Rule proposal: `no-null`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#612]
https://github.com/sindresorhus/meta/issues/7

# Fail

```js
const foo = null;
```

# Pass

```js
let foo;
```

Exclude `Object.create`

```js
let foo = Object.create(null);
```

- There is already a plugin do this, https://github.com/nene/eslint-plugin-no-null/blob/master/index.js
- Rule proposal: no-null in `ESLint` project https://github.com/eslint/eslint/issues/6701
- TypeScript Coding guidelines https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined

**Comment by himynameisdave** [comment]
While I wouldn't personally use this rule (as I prefer to use `null` instead of `undefined`, [sorry Crockford](https://www.youtube.com/watch?v=PSGEjv3Tqo0&t=9m21s)), I can start a PR for this.

Wondering if we want to make it configurable, so perhaps `prefer-null-or-undefined`, where the user can specify which they prefer (but not both), but for now I will go forward with just `no-null`.

**Comment by sindresorhus** [comment]
I would prefer to keep it to only prevent `null`.
